agent/src/beerocks/slave/backhaul_manager/internal/expected_ap_metrics_response.h

```diff
@@ -0,0 +1,54 @@
+/* SPDX-License-Identifier: BSD-2-Clause-Patent
+ *
+ * SPDX-FileCopyrightText: 2016-2020 the prplMesh contributors (see AUTHORS.md)
+ *
+ * This code is subject to the terms of the BSD+Patent license.
+ * See LICENSE file for more details.
+ */
+
+#ifndef _EXPECTED_AP_METRICS_RESPONSE_H
+#define _EXPECTED_AP_METRICS_RESPONSE_H
+
+#include <bcl/beerocks_defines.h>
+#include <tlvf/CmduMessageTx.h>
+#include <tlvf/common/sMacAddr.h>
+#include <tlvf/tlvftypes.h>
+#include <unordered_set>
+
+class ExpectedApMetricsResponse {
+public:
+    /**
+     * @brief resets this object with a new mid
+     * @details resets all members of this object to its empty initial state except for the mid 
+     * that is set as was given bt the parameter (mid default value is zero (0))
+     * @param mid - the new mid value
+     * @return radio mac of the found slave if found, otherwise empty string
+     */
+    void reset_to_new_mid(uint16_t mid = 0);
+    uint16_t get_mid() const;
+    ieee1905_1::CmduMessageTx &create_tx_message();
+
+    // bssid related operations
+public:
```

Comment:
Why another `public` section?

Reply:
Removed.
However -
Like always: **maintainability**.
Having the class header divided into sections makes it _much_ more readable when maintaining the code, when using the code. It is an additional information for the reader about the intentions of the writer, the way she sees the internal structure of the class. 
In this example, the `public` separator also has a comment above it "bssid related operations" which means that it is easier to rational behind the intent of the writer.